example/src/screens/CreateRoom.tsx

```diff
@@ -0,0 +1,267 @@
+        maxBandwidth: { l: 150, m: 500, h: 1500 },
+        videoTrackMetadata: { active: true, type: 'camera' },
+        audioTrackMetadata: { active: true, type: 'audio' },
+        isSpeakerphoneOn: false,
+      });
+      await joinRoom();
+      navigation.navigate('Room');
+    } catch (err) {
+      console.warn(err);
+    }
+  }, [
+    requestPermissions,
+    mbConnect,
+    joinRoom,
+    roomName,
+    isSimulcastOn,
+    username,
+    SERVER_URL,
+  ]);
+
+  return (
+    <ScrollView
+      contentContainerStyle={{ flexGrow: 1 }}
+      keyboardShouldPersistTaps="handled"
+    >
+      <KeyboardAvoidingView
+        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+        style={{
+          flex: 1,
+          width: '100%',
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+        enabled
+        keyboardVerticalOffset={height}
+      >
+        <BackgroundWrapper>
+          <Modal
+            headline="Discard meeting"
+            body="Are you sure you want to discard creation of this meeting?"
+            visible={isModalVisible}
+            onClose={() => setIsModalVisible(false)}
+          >
+            <StandardButton
+              type="danger"
+              onPress={() => {
+                setIsModalVisible(false);
+                setRoomName('');
+                setUsername('');
+                navigation.dispatch(modalAction!);
+              }}
+            >
+              Yes, discard meeting
+            </StandardButton>
+          </Modal>
+          <View style={styles.inner}>
+            <View>
+              <Typo variant="h4">Videoconferencing for everyone</Typo>
+            </View>
+
+            <View style={styles.smallTitle}>
+              <Typo variant="chat-regular">
+                Create a new room to start the meeting
+              </Typo>
+            </View>
+            <View style={styles.roomInputLabel}>
+              <Typo variant="body-small">Room name</Typo>
+            </View>
+
+            <View style={styles.roomInput}>
+              <TextInput
+                placeholder="Room name"
+                value={roomName}
+                onChangeText={(val) => {
+                  setRoomName(val);
+                }}
+              />
+            </View>
+            <View style={styles.usernameInputLabel}>
+              <Typo variant="body-small">Your name</Typo>
+            </View>
+            <View style={styles.usernameInput}>
+              <TextInput
+                placeholder="Your name"
+                value={username}
+                onChangeText={(val) => {
+                  setUsername(val);
+                }}
+              />
+            </View>
+          </View>
+          <View style={{ flex: 1 }} />
+
+          <View style={styles.createRoomButton}>
+            <StandardButton
+              onPress={() => {
+                TrimTrailingSpacesForTextInputs();
```

Comment:
this is not going to work :p 
`TrimTrailingSpacesForTextInputs` changes the state (`setRoomName`). State changes are asynchronous so when we read the room name in the `connect` function we'll get the old value with spaces (for more explanation see [here](https://react.dev/reference/react/useState#ive-updated-the-state-but-logging-gives-me-the-old-value))

Reply:
Changed function to be async and added `await`.